### ISSUE TITLE
`fixture_union()` able to take a list of fixtures instead of a list of values

### ISSUE THREAD

**Issue by smarie** [smarie/python-pytest-cases#34]
This seems like the missing piece. When we have this feature, combining it with `param_fixture` / `param_fixtures` will enable us to reproduce any of the scenarios mentioned in 
https://github.com/pytest-dev/pytest/issues/349

Note that we could also provide a helper goodie `@pytest_parametrize_plus(..., fixtures=[...])` to do the same but directly on a function.

**Comment by smarie** [comment]
A few notes after a couple hours on this topic: I managed to find a way using the hooks, to change the CallSpecs of the tests, that is, the list of calls that are to be made (together with the associated parameter values).

However now I'm stuck with this prototypical example

```python
from pytest_cases import param_fixture, fixture_union

a = param_fixture('a', [0, 1])
b = param_fixture('b', [2, 3])
c = fixture_union('c', a, b)

def test_fixture_union(c, a):
    print(c, a)
```

What should be the correct list of calls to `test_fixture_union` ?

**Comment by smarie** [comment]
Update: first implementation seems to work correctly:

![image](https://user-images.githubusercontent.com/3236794/55184646-fb9dab00-5192-11e9-9e85-2035036e6e08.png)

**Comment by smarie** [comment]
It seems that my second example was wrong in terms of expected results. Indeed I forgot the `a`.
This seems to be the appropriate list:

```bash
test_fixture_union[x-1]  # c=a, a='x', d=b, b=1
test_fixture_union[x-2]  # c=a, a='x', d=b, b=2
test_fixture_union[x]     # c=a, a='x', d=a
test_fixture_union[y-1]  # c=a, a='y', d=b, b=1
test_fixture_union[y-2]  # c=a, a='y', d=b, b=2
test_fixture_union[y]     # c=a, a='y', d=a
test_fixture_union[1-x]  # c=b, b=1, a='x', d=b
test_fixture_union[1-x]  # c=b, b=1, a='x', d=a
test_fixture_union[1-y]  # c=b, b=1, a='y', d=b
test_fixture_union[1-y]  # c=b, b=1, a='y', d=a
test_fixture_union[2-x]  # c=b, b=2, a='x', d=b
test_fixture_union[2-x]  # c=b, b=2, a='x', d=a
test_fixture_union[2-y]  # c=b, b=2, a='y', d=b
test_fixture_union[2-y]  # c=b, b=2, a='y', d=a
```

By seeing this list we clearly see that ids need to contain something indicating which fixture was selected by the union.

**Comment by smarie** [comment]
New version works as expected. Ready to merge into master

![image](https://user-images.githubusercontent.com/3236794/55479570-a6441c80-561e-11e9-9b1d-6fd693d5ae9f.png)

**Comment by smarie** [comment]
Fixed in 6e2f3c1f6291b494cd4d3ff3fc208ddadfab5419
